types/dadata/tslint.json

```
@@ -0,0 +1,6 @@
+{
+  "extends": "@definitelytyped/dtslint/dt.json",
+  "rules": {
+    "export-just-namespace": false
+  }
```

```suggestion
  "extends": "@definitelytyped/dtslint/dt.json"
```
add this rule inline, if still required

---

@peterblazejewicz i'm sorry, but build cant get completed, if i remove "export-just-namespace": false

---

hi,

so here is a code from readme (tests.ts here)
```ts
import DaData = require('dadata');

const dadata_account_1 = new DaData('<API_KEY_1>', '<SECRET_KEY_1>');
const dadata_account_2 = new DaData('<API_KEY_2>', '<SECRET_KEY_2>');
dadata_account_1('address', ['мск сухонска 11/-89'], (err, res) => {});
```
and this is sample implementation missing your current code. It seems you've just added interfaces for models, nothing for actually using export symbols (which are a function with constructor function), something like:

```ts
...

        source: null;
    }

    interface Results {
        // TBD
    }

    interface Service {
        (
            apiType: string,
            queries: string[],
            callback: (err: Error | undefined, res: Results | undefined) => void,
        ): void;
    }
}

interface DaDataConstructor {
    new (apiKey: string, secretKey: string): DaData.Service;
    (apiKey: string, secretKey: string): DaData.Service;
}

declare const DaData: DaDataConstructor;

export = DaData;
```

---

@peterblazejewicz its my fault - i attempted to add definitions for api from service dadata (https://dadata.ru/api/), i dont saw package "dadata" exist in npm )) Ive been already renamed my module - can you check it ?

---

@peterblazejewicz my definitions only for responses from https://dadata.ru/api/, not for any existing package in npm )

---

ops, not sure why dadata got into the discussion, please disregard my commens and sample as not relevant